service/hub/server/server.go

```diff
@@ -358,3 +369,26 @@ func (as *AuthServer) Stop() {
 func joseBase64UrlEncode(b []byte) string {
 	return strings.TrimRight(base64.URLEncoding.EncodeToString(b), "=")
 }
+
+func (as *AuthServer) pullLimitCheck(ar *AuthRequest) bool {
+	if len(ar.Scopes) == 0 {
+		return true
+	}
+	for _, cidr := range as.config.Server.WhiteIPCidrList {
+		if utils.IsIPInCIDR(ar.RemoteIP, cidr) {
+			return true
+		}
+	}
+	for _, user := range as.config.Server.WhiteUserList {
+		if ar.Account == user {
+			return true
+		}
+	}
+
+	// nomal user pull/push request
+	as.PullReqCounter.Increment(ar.RemoteIP.String())
+	as.PullReqCounter.Increment(ar.Account)
+
+	return as.PullReqCounter.Get(ar.RemoteIP.String()) < as.config.Server.MaxRequestsPerIP &&
+		as.PullReqCounter.Get(ar.Account) < as.config.Server.MaxRequestsPerAccount
```

Comment:
can you use ratelimter pkg instead of writing it ?

Reply:
ok

Reply:
One question about `golang.org/x/time/rate` (not sure if it is the pkg): I have to mantain a limiter for each user and ip, it's a huge cost since user and ip will increace.

Reply:
I think  the golang.org/x/time/rate is enough for  now , you can controller the total cost by limiter .

Reply:
ok, thx for advice.